### PR TITLE
Test real inputs to ns delta

### PR DESCRIPTION
**Pull request by immesys**
Ben made a good point: it is convenient to specify floats to ns delta. Unfortunately it won't function as expected if you do, so we either need to deal with that (by casting earlier, before overflow) or we need to maybe error out if a float is received.

Here is a failing test to show what I mean

### CODE REVIEW

File: btrdb/utils/timez.py [hunk 1]
```diff
@@ -156,11 +159,15 @@ def ns_delta(days=0, hours=0, minutes=0, seconds=0, milliseconds=0, \
...
+        raise TypeError("nanoseconds argument must be an integer")
+
+    units = []
+    for unit in (days, hours, minutes, seconds, milliseconds, microseconds):
+        if isinstance(unit, float):
+            unit = Decimal(unit)
+        units.append(unit)
 
+    factors = [DAY, HOUR, MINUTE, SECOND, MILLESECOND, MICROSECOND]
+    nanoseconds += sum(map(mul, units, factors))
```
Comment (bbengfort) [inline btrdb/utils/timez.py]: @immesys you're absolutely right, I wasn't thinking through the overflow case when days, hours, minutes could be floats (and I probably did my back of the napkin math on overflow for seconds incorrectly). 

anyhoo ... here's is a proposal that fixes the test you posted, using the better precision of Decimal in the case of float, assuming `int` otherwise. 

@looselycoupled I slapped this together and didn't think through any other edge cases; what do you think about this?

File: btrdb/utils/timez.py [hunk 2]
```diff
@@ -156,11 +159,15 @@ def ns_delta(days=0, hours=0, minutes=0, seconds=0, milliseconds=0, \
...
+    if not isinstance(nanoseconds, int):
+        raise TypeError("nanoseconds argument must be an integer")
+
+    units = []
+    for unit in (days, hours, minutes, seconds, milliseconds, microseconds):
+        if isinstance(unit, float):
+            unit = Decimal(unit)
+        units.append(unit)
 
+    factors = [DAY, HOUR, MINUTE, SECOND, MILLESECOND, MICROSECOND]
```
Comment (immesys) [inline btrdb/utils/timez.py]: While you are here can you fix the MILL**E**SECOND :p

Comment (bbengfort) [reply btrdb/utils/timez.py]: Je ne sais pas ce que vous dites, ceci est orthographié correctement!

... hehe, actually I just checked even in French the e is at the end!

File: tests/btrdb/utils/test_timez.py [hunk 3]
```diff
@@ -174,15 +174,21 @@ def test_ns_delta(self):
         """
         Assert ns_delta converts inputs properly
         """
-        val = ns_delta(1,1,1,1,1,1,1)
-        assert val == int(1 + 1e3 + 1e6 + 1e9 + (1e9 * 60)  + (1e9 * 60 * 60) + \
-            (1e9 * 60 * 60 * 24) )
+        val = ns_delta(1,2,1,3,1,23,1)
```
Comment (bbengfort) [inline tests/btrdb/utils/test_timez.py]: Changed this to a non-symmetric value because I was worried I had screwed up the arg order.

Comment (immesys) [reply tests/btrdb/utils/test_timez.py]: Good call

**Comment by immesys** [comment]
@looselycoupled can you merge after you review?

File: btrdb/utils/timez.py (hunk 1 above)
Comment (looselycoupled) [reply btrdb/utils/timez.py]: Looks great to me. Will try and think of edge cases...
